lmfit/minimizer.py

```diff
@@ -294,8 +294,10 @@ def penalty(self, params):
             objective function is an array, return the array sum-of-squares
         """
         r = self.__residual(params)
-        if isinstance(r, ndarray):
+        if isinstance(r, ndarray) and likelihood_fun is None:
```

Comment:
As I understand it, this function gets called many times during the optimization?  Probably worth bench marking these method, I was worried about creating a new lambda every time through.